app/views/videos/index.html.haml

```diff
@@ -3,6 +3,6 @@
     %header
       %h3= link_to category.name, category_path(category)
     .videos.row
-      - cat.videos.each do |video|
+      - category.recent_videos.each do |video|
         .video.col-sm-2
           = link_to image_tag( video.small_cover_url), video_path(video)
```

Comment:
Eventually you can call `video` instead of `video_path(video)` here and it will do nasty rails magic to produce correct route :)